### PR TITLE
[openstack] fix floating ip attaching

### PR DESCRIPTION
**Pull request by vills**
- Check that floating IP already attached;
- Fix floating IP attaching.

I really don't know how it work previously, cause in fog_driver.rb attach_ip() executes in different ways from attach_*() functions.

### CODE REVIEW

File: lib/chef_metal_fog/providers/openstack.rb [hunk 1]
```diff
@@ -32,6 +32,45 @@ def self.compute_options_for(provider, id, config)
         [result, id]
       end
 
+      # Check given IP already attached and attach if necessary
+      def attach_floating_ips(action_handler, machine_spec, machine_options, server)
+        if option_for(machine_options, :floating_ip_pool)
+          Chef::Log.info 'Attaching IP from pool'
+          action_handler.perform_action "attach floating IP from #{option_for(machine_options, :floating_ip_pool)} pool" do
+            pool = option_for(machine_options, :floating_ip_pool)
```
Comment (tyler-ball) [inline lib/chef_metal_fog/providers/openstack.rb]: Since you reference this multiple times, you can move it to the top of the method definition.

**Comment by jkeiser** [comment]
@tyler-ball we have confirmation in #77 that merging this fixes OpenStack.  It may be worth making sure AWS still works (comments there indicate there may be an issue), but I don't think that should block this since it shouldn't make AWS any *worse* if it's not working.

**Comment by Sliim** [comment]
Hmm, I have another issue when attempting to attach floating ip to my instance.
I merged this change in the HEAD, and yeah #77 is fixed for OpenStack. But now i'm getting a BadRequest on API call when the driver attempt to attach the ip to the server, the error message is `No nw_info cache associated with instance`.

I suspect the instance is not fully created and doesn't contains any metadata.. So the addFloatingIp action fails. My temporary (and sad) workaround is to sleep 10s before `compute.associate_address` call in `attach_ip` method (provisioning/fog_driver/providers/openstack.rb:73).

**Comment by tyler-ball** [comment]
Closing this in favor of https://github.com/chef/chef-provisioning-fog/pull/86 - please move all conversation there!

**Pull request closed without merging**